postgresql/resource_postgresql_publication.go

```diff
@@ -372,13 +372,10 @@ func resourcePostgreSQLPublicationReadImpl(db *DBConnection, d *schema.ResourceD
 		}
 		tables = append(tables, table)
 	}
-
-	switch {
-	case err == sql.ErrNoRows:
-		log.Printf("[WARN] No PostgreSQL tables found for Publication %s", PublicationName)
-	case err != nil:
-		return fmt.Errorf("Error reading Publication tables: %w", err)
+	if err := rows.Err(); err != nil {
```

Comment:
I added this error handling here to catch any error that can happen during the rows.Next iteration, as described in the  doc https://pkg.go.dev/database/sql#Rows.Err